cluster/images/conformance/run_e2e.sh

```diff
@@ -39,23 +39,28 @@ saveResults() {
 # We get the TERM from kubernetes and handle it gracefully
 trap shutdown TERM
 
-ginkgo_args=(
-    "--focus=${E2E_FOCUS}"
-    "--skip=${E2E_SKIP}"
-    "--noColor=true"
-)
-
+ginkgo_args=()
 if [[ -n ${E2E_DRYRUN:-} ]]; then
     ginkgo_args+=("--dryRun=true")
 fi
 
 case ${E2E_PARALLEL} in
-    'y'|'Y')           ginkgo_args+=("--nodes=25") ;;
-    [1-9]|[1-9][0-9]*) ginkgo_args+=("--nodes=${E2E_PARALLEL}") ;;
+    'y'|'Y'|'true')
+        # The flag '--p' will automatically detect the optimal number of ginkgo nodes.
+        ginkgo_args+=("--p")
+        # Skip serial tests if parallel mode is enabled.
+        E2E_SKIP="\\[Serial\\]|${E2E_SKIP}" ;;
```

Comment:
This is somewhat naive. It means any conformance tests marked with `[Serial]` won't be run, which reduces the value of the conformance suite. That being said, I have noticed that the number of such tests has significantly reduced (if i remember correctly, in kubernetes 1.11.x it was around 7).

Reply:
You cannot safely run `[Serial]` tagged tests in parallel, that's what the tag means. ginkgo nodes must be zero to run these.

Reply:
I think "naive" was probably a bad choice of words, I was actually thinking of "misleading". It's misleading because it allows users to think they have run all conformance tests, when in reality there may be some `[Serial]` tests that haven't been run. Maybe I'm just overreacting though. Anyway, previously for CI purposes I would run `sonobuoy` twice -- once for all the tests that could be run in parallel, and again for all the tests that had to be run serially.

Reply:
For submitting conformance results the standard is one run in non-serial mode / a single ginkgo node with a specific config, but doing something like this is definitely very useful for CI results. :+1: